apps/root/src/services/earnService.ts

```diff
@@ -506,18 +500,20 @@ export class EarnService extends EventsManager<EarnServiceData> {
       throw new Error('Could not find strategy');
     }
 
-    let permissionPermit: Awaited<ReturnType<typeof this.getSignatureForPermission>> | undefined;
-    const hasPermission = await this.companionHasPermission(earnPositionId, EarnPermission.WITHDRAW);
-
-    if (!hasPermission) {
-      const companionAddress = this.contractService.getEarnCompanionAddress(strategy.farm.chainId);
-      // eslint-disable-next-line @typescript-eslint/no-unused-vars
-      permissionPermit = await this.getSignatureForPermission({
-        earnPosition: userStrategy,
-        chainId: strategy.farm.chainId,
-        contractAddress: companionAddress,
-        permission: EarnPermission.WITHDRAW,
-      });
+    if (requirePermit) {
+      let permissionPermit: Awaited<ReturnType<typeof this.getSignatureForPermission>> | undefined;
+      const hasPermission = await this.companionHasPermission(earnPositionId, EarnPermission.WITHDRAW);
+
+      if (!hasPermission) {
+        const companionAddress = EARN_COMPANION_ADDRESS[strategy.farm.chainId];
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        permissionPermit = await this.getSignatureForPermission({
```

Comment:
aren't we already signing this?

Reply:
I think this comment is on outdated code, now the signature is handleted in the txSteps